packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -197,19 +197,27 @@ - (void)addPayment:(FlutterMethodCall *)call result:(FlutterResult)result {
 }
 
 - (void)finishTransaction:(FlutterMethodCall *)call result:(FlutterResult)result {
-  if (![call.arguments isKindOfClass:[NSString class]]) {
+  if (![call.arguments isKindOfClass:[NSDictionary class]]) {
     result([FlutterError errorWithCode:@"storekit_invalid_argument"
-                               message:@"Argument type of finishTransaction is not a string."
+                               message:@"Argument type of finishTransaction is not a Dictionary"
                                details:call.arguments]);
     return;
   }
-  NSString *transactionIdentifier = call.arguments;
+  NSDictionary *paymentMap = (NSDictionary *)call.arguments;
+  NSString *transactionIdentifier = [paymentMap objectForKey:@"transactionIdentifier"];
+  NSString *productIdentifier = [paymentMap objectForKey:@"productIdentifier"];
 
   NSArray<SKPaymentTransaction *> *pendingTransactions =
       [self.paymentQueueHandler getUnfinishedTransactions];
 
   for (SKPaymentTransaction *transaction in pendingTransactions) {
-    if ([transaction.transactionIdentifier isEqualToString:transactionIdentifier]) {
+    // If the user cancels the purchase dialog we won't have a transactionIdentifier. So if it is
```

Comment:
I think these two words break the line. Same below.

Reply:
I reformatted the comment